Source/Core/InputCommon/ControllerInterface/WGInput/WGInput.cpp

```diff
@@ -668,6 +684,36 @@ static void RemoveDevice(const WGI::RawGameController& raw_game_controller)
 // (H is the lambda)
 #pragma warning(disable : 4265)
 
+static void HandleAddRemoveEvent(AddRemoveEvent evt)
+{
+  try
+  {
+    winrt::init_apartment();
+  }
+  catch (const winrt::hresult_error& ex)
+  {
+    ERROR_LOG_FMT(CONTROLLERINTERFACE,
+                  "WGInput: Failed to CoInitialize for add/remove controller event: {}",
+                  WStringToUTF8(ex.message()));
+    return;
+  }
+  Common::ScopeGuard coinit_guard([] { winrt::uninit_apartment(); });
+
+  switch (evt.type)
+  {
+  case AddRemoveEventType::Add:
```

Comment:
Minor, but maybe we should call this: `AddOrReplace`, given it's removing any prior version of the same controller ID.